scrapypuppeteer/middleware.py

```diff
@@ -29,95 +30,90 @@
     PuppeteerJsonResponse,
 )
 from scrapypuppeteer.request import ActionRequest, PuppeteerRequest, CloseContextRequest
+from scrapypuppeteer.scrappypyppeteer import LocalScrapyPyppeteer
 
 
-class PuppeteerServiceDownloaderMiddleware:
-    """
-    This downloader middleware converts PuppeteerRequest instances to
-    Puppeteer service API requests and then converts its responses to
-    PuppeteerResponse instances. Additionally, it tracks all browser contexts
-    that spider uses and performs cleanup request to service right before
-    spider is closed.
+class BrowserManager(ABC):
+    @abstractmethod
+    def process_request(self, request, spider):
+        pass
+    
+    @abstractmethod
+    def close_used_contexts(self):
+        pass
 
-        Additionally, the middleware uses these meta-keys, do not use them, because their changing
-    could possibly (almost probably) break determined behaviour:
-    'puppeteer_request', 'dont_obey_robotstxt', 'proxy'
 
-    Settings:
+class LocalBrowserManager(BrowserManager):
+    def __init__(self):
+        self.local_scrapy_pyppeteer = LocalScrapyPyppeteer()
 
-    PUPPETEER_SERVICE_URL (str)
-    Service URL, e.g. 'http://localhost:3000'
+    def process_request(self, request):
+        
+        action_request = ActionRequest(
+            url='http://_running_local_',
```

Comment:
Why we need this proxy class for handling local requests?

Reply:
I have fixed everything except this part. Yes, this class might be redundant, but it is convenient to use and all the code is already written for it. I have removed unnecessary data from the constructor, but it would be very difficult to rewrite everything, and I'm not sure if it would make sense.